drivers/ethernet/eth_w5500.c

```diff
@@ -342,28 +342,52 @@ static int w5500_set_config(const struct device *dev,
 {
 	uint8_t mode;
 	uint8_t mr = W5500_S0_MR_MF;
+	struct w5500_runtime *ctx = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
```

Comment:
I've been looking at this and I'm wondering why place this switch here instead of implementing it on the `generate_mac` callback on the `w5500_runtime` struct. It seems to mix the mode with the MAC attribution.

Reply:
It seems that when writing the comment marking the issue resolved then the comment itself is lost. This was my answer:
`generate_mac` is private field and used to assign random MAC. Random MAC that changes on every restart was something that I wanted to avoid in my application. In my solution I'm storing the MAC address in application and updating the driver MAC address on application startup.